metadata-service/auth-filter/src/main/java/com/datahub/authentication/filter/AuthenticationFilter.java

```diff
@@ -155,7 +166,7 @@ private void buildAuthenticatorChain() {
       final NoOpAuthenticator noOpAuthenticator = new NoOpAuthenticator();
       noOpAuthenticator.init(ImmutableMap.of(
           SYSTEM_CLIENT_ID_CONFIG,
-          this.configurationProvider.getAuthentication().getSystemClientId()));
+          this.configurationProvider.getAuthentication().getSystemClientId()), null);
```

Comment:
I don't like passing null into this init method. Very easy to have aan NPE.

Please use Collections.emptyMap() if you don't have anything to pass.